zha/zigbee/device.py

```diff
@@ -869,16 +887,32 @@ async def async_initialize(self, from_cache: bool = False) -> None:
         # At this point we can compute a primary entity
         self._compute_primary_entity()
 
+        # Sync the device's firmware version with the first platform entity
+        for (platform, _unique_id), entity in self.platform_entities.items():
+            if platform != Platform.UPDATE:
+                continue
```

Comment:
Seems a bit weird to go from entity level back up to device level for this info. We create a small (unnecessary?) dependence here.

Alternative might be for the `device.firmware_version` to return the cached `current_file_version` attribute (like the update entity currently does) and also have the device subscribe to `current_file_version` attribute changes (instead of the update entity doing that).
Then, you would have the update entity use `device.firmware_version` and have it subscribe to the `DeviceUpdatedEvent` events.

Reply:
Good point. I was thinking of this from the opposite direction: the update entity can be overriden per-device and then provide a formatted firmware version (without the ZHA device needing to be patched). It would avoid having the device scan through clusters to determine the correct firmware version.

Reply:
> the update entity can be overriden per-device and then provide a formatted firmware version (without the ZHA device needing to be patched)

Does the base ZHA device need to know the "friendly version" though? For HA, we generally use the update entity for that, right? Except for the little firmware version string in the device card/registry...

But yeah, I'm fine with this as well, so feel free to merge as-is if you want.